wweval/R/ms_fig5.R

```diff
@@ -0,0 +1,465 @@
+#' full distribution
+#' @export
+#'
+make_fig5_hub_performance <- function(all_scores,
+                                      cfa_real_time_scores,
+                                      figure_file_path,
+                                      all_time_period,
+                                      real_time_period,
+                                      models_to_show,
+                                      summarize_across_horizon = FALSE,
+                                      baseline_model = "COVIDhub-4_week_ensemble") {
+  subset_scores <- all_scores |>
+    dplyr::filter(model %in% !!models_to_show)
+
+  if (isTRUE(summarize_across_horizon)) {
+    scores_by_model_all_time <- subset_scores |>
+      data.table::as.data.table() |>
+      scoringutils::summarise_scores(
+        by = c("model", "forecast_date", "location", "horizon")
+      ) |>
+      dplyr::mutate(
+        period = {{ all_time_period }}
+      )
+
+    scores_by_model_real_time <- subset_scores |>
+      dplyr::filter(forecast_date >= lubridate::ymd("2024-02-05")) |>
+      dplyr::bind_rows(cfa_real_time_scores) |>
+      data.table::as.data.table() |>
+      scoringutils::summarise_scores(
+        by = c("model", "forecast_date", "location")
+      ) |>
+      dplyr::mutate(
+        period = {{ real_time_period }}
+      )
+  } else {
+    scores_by_model_all_time <- subset_scores |>
+      data.table::as.data.table() |>
+      dplyr::mutate(
+        period = {{ all_time_period }}
+      )
+    scores_by_model_real_time <- subset_scores |>
+      data.table::as.data.table() |>
+      dplyr::filter(forecast_date >= lubridate::ymd("2024-02-05")) |>
+      dplyr::bind_rows(cfa_real_time_scores) |>
+      dplyr::mutate(
+        period = {{ real_time_period }}
+      )
+  }
+
+  scores <- dplyr::bind_rows(
+    scores_by_model_all_time,
+    scores_by_model_real_time
+  )
+
+  # Want to get the mean across all forecast dates and locations for each
+  # model during each period
+  mean_scores <- scores |>
+    scoringutils::summarise_scores(
+      by = c("model", "period")
+    ) |>
+    dplyr::rename(
+      mean_score = interval_score
+    ) |>
+    dplyr::select(
+      model, period, mean_score
+    )
+
+  baseline_scores <- scores |>
+    dplyr::filter(model == {{ baseline_model }}) |>
+    dplyr::select(location, forecast_date, horizon, interval_score) |>
+    dplyr::rename(baseline_score = interval_score)
+
+  scores_final <- scores |>
+    dplyr::left_join(mean_scores, by = c("model", "period")) |>
+    dplyr::left_join(baseline_scores, by = c(
+      "forecast_date", "horizon",
+      "location"
+    )) |>
+    dplyr::mutate(relative_wis = interval_score / baseline_score) |>
+    dplyr::filter(model != {{ baseline_model }}) |>
+    dplyr::mutate(
+      fig_order =
+        dplyr::case_when(
+          period == "Feb 2024-Mar 2024" ~ 1,
+          period == "Oct 2023-Mar 2024" ~ 0
+        ),
+      horizon = forcats::fct_reorder(period, fig_order)
+    )
```

Comment:
As above, should handle this with an `order_period` function that uses base `factor`, not `forecats::fct_reorder()`

Reply:
This is done!